src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,598 @@ const ActorInit En_Box_InitVars = {
+            }
+        }
+    }
+}
+
+void func_809C9EF8(EnBox* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f vel = { 0.0f, 1.0f, 0.0f };
+    Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    f32 f0;
+
+    this->iceSmokeTimer++;
+    func_8002F974(&this->dyna.actor, NA_SE_EN_MIMICK_BREATH - SFX_FLAG);
+    if (Math_Rand_ZeroOne() < 0.3f) {
+        f0 = 2.0f * Math_Rand_ZeroOne() - 1.0f;
+        pos = this->dyna.actor.posRot.pos;
+        if (this->type == ENBOX_TYPE_SMALL || this->type == ENBOX_TYPE_6 || this->type == ENBOX_TYPE_ROOM_CLEAR_SMALL ||
+            this->type == ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL) {
+            pos.x += f0 * 10.0f * Math_Sins(this->dyna.actor.posRot.rot.y + 0x4000);
+            pos.z += f0 * 10.0f * Math_Coss(this->dyna.actor.posRot.rot.y + 0x4000);
+            f0 = 2.0f * Math_Rand_ZeroOne() - 1.0f;
+            vel.x = f0 * 0.8f * Math_Sins(this->dyna.actor.posRot.rot.y);
+            vel.y = 1.8f;
+            vel.z = f0 * 0.8f * Math_Coss(this->dyna.actor.posRot.rot.y);
+        } else {
+            pos.x += f0 * 20.0f * Math_Sins(this->dyna.actor.posRot.rot.y + 0x4000);
+            pos.z += f0 * 20.0f * Math_Coss(this->dyna.actor.posRot.rot.y + 0x4000);
+            f0 = 2.0f * Math_Rand_ZeroOne() - 1.0f;
+            vel.x = f0 * 1.6f * Math_Sins(this->dyna.actor.posRot.rot.y);
+            vel.y = 1.8f;
+            vel.z = f0 * 1.6f * Math_Coss(this->dyna.actor.posRot.rot.y);
+        }
+        func_8002AA44(globalCtx, &pos, &vel, &accel, 150);
+    }
+}
+
+void EnBox_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBox* this = THIS;
+
+    if (this->movementFlags & ENBOX_MOVE_STICK_TO_GROUND) {
+        this->movementFlags &= ~ENBOX_MOVE_STICK_TO_GROUND;
+        EnBox_ClipToGround(this, globalCtx);
+    }
+
+    (this->actionFunc)(this, globalCtx);
+
+    if (!(this->movementFlags & ENBOX_MOVE_IMMOBILE)) {
+        Actor_MoveForward(&this->dyna.actor);
+        func_8002E4B4(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 0x1C);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/EnBox_Init.s")
+    switch (this->type) {
+        case ENBOX_TYPE_SMALL:
+        case ENBOX_TYPE_6:
+        case ENBOX_TYPE_ROOM_CLEAR_SMALL:
+        case ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL:
+            Actor_SetHeight(&this->dyna.actor, 20.0f);
+            break;
+        default:
+            Actor_SetHeight(&this->dyna.actor, 40.0f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/EnBox_Destroy.s")
+    if ((this->dyna.actor.params >> 5 & 0x7F) == 0x7C && this->actionFunc == EnBox_Open &&
+        this->skelanime.animCurrentFrame > 45 && this->iceSmokeTimer < 100) {
+        func_809C9EF8(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C92F4.s")
+void EnBox_PostLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
```

Comment:
The type is called "2" but that's just because there are 2 slightly different variants, so we usually just name the function without it.
```suggestion
void EnBox_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
```